Tests pass, 197/564. Ran phpunit standalone — `jp` is unusable in my environment, `pnpm install` fails on a lockfile supply-chain policy check (`ERR_PNPM_TRUST_DOWNGRADE`, 40 `@wordpress/*` entries missing publish times). Unrelated to this PR.

Mutation check reproduces: removing the Tracks block fails exactly one test.

Ordering claim checks out — `jpTracksContext.blog_id` comes from `Connection_Initial_State::render()` via `render_connection_initial_state()` on `admin_print_scripts` pri 1 (head), before the footer `jp-tracks-functions`.

### 1. `reset_scripts()` doesn't dequeue — one assertion can't fail

`Admin_Modernization_Gating_Test.php:196`. `wp_deregister_script()` → `WP_Dependencies::remove()` unsets `$registered` only; `$wp_scripts->queue` keeps the handle. `wp_script_is( $h, 'enqueued' )` reads `queue`.

Proven: with the production Tracks block removed, `allow_analytics()` added to the legacy test, and line 126 dropped, the file still goes 8/8 green. Line 127's `enqueued` assertion is carried entirely by line 126's `registered` one. Add to `reset_scripts()`:

```php
wp_dequeue_script( $handle );
wp_deregister_script( $handle );
```

### 2. Changelog overstates the change

"Send Tracks analytics from the modernized dashboard." — nothing sends events yet, as the description says. Suggest "Register the Tracks client on the modernized dashboard."

### 3. Pre-existing, flag-flip risk

Flag on + `build/build.php` absent → `add_wp_admin_submenu()` falls back to `plugin_settings_page()`, `enqueue_admin_scripts()` still returns before the legacy bundle. Blank page. Not introduced here, but this PR makes it untestable, so worth a build-presence check in `is_modernized()` before the flip.

### Open questions

1. Agreed — initial state was already walked back for the same reason. Needs @dhasilva.
2. Mock is contained; no other test in the package calls either method. Don't add a production seam for it.
3. Agreed on `RunInSeparateProcess`.

### Nit

Trim the comments. 229-line test file, 20-line mock for one empty function, a 5-line comment where the 3-line one it replaced said enough, and a description with three numbered caveats. Findings 1–3 above are ~15 lines; the same density would cut this PR by a third and make the real caveats easier to see.

— Tangerine

--- a/projects/packages/backup/changelog/add-backup-modernized-tracks
+++ b/projects/packages/backup/changelog/add-backup-modernized-tracks
@@ -1,0 +1,4 @@
+Significance: patch
+Type: added
+
+Send Tracks analytics from the modernized dashboard.

--- a/projects/packages/backup/src/class-jetpack-backup.php
+++ b/projects/packages/backup/src/class-jetpack-backup.php
@@ -250,9 +250,17 @@ class Jetpack_Backup {
 				wp_enqueue_script( 'wp-jp-i18n-loader' );
 			}
 
-			// wp-build manages its own enqueue pipeline. The legacy script,
-			// initial state, and tracking are intentionally skipped for the
-			// wp-build dashboard.
+			// Tracks needs its client callables (`jpTracks.recordEvent`) on the
+			// page before any dashboard component can record an event. The
+			// esbuild bundles don't declare them as a dependency either, so
+			// enqueue them here, before the early return, on the same
+			// `can_use_analytics()` gate the legacy path uses below.
+			if ( self::can_use_analytics() ) {
+				Tracking::register_tracks_functions_scripts( true );
+			}
+
+			// wp-build manages its own enqueue pipeline. The legacy script and
+			// its initial state are skipped for the wp-build dashboard.
 			return;
 		}
 

--- a/projects/packages/backup/tests/php/Admin_Modernization_Gating_Test.php
+++ b/projects/packages/backup/tests/php/Admin_Modernization_Gating_Test.php
@@ -15,8 +15,11 @@ use PHPUnit\Framework\Attributes\CoversClass;
 use PHPUnit\Framework\TestCase;
 use WorDBless\Options as WorDBless_Options;
 use function add_filter;
+use function has_action;
+use function remove_action;
 use function remove_all_actions;
 use function remove_all_filters;
+use function set_current_screen;
 use function wp_dequeue_script;
 use function wp_deregister_script;
 use function wp_script_is;
@@ -38,6 +41,12 @@ class Admin_Modernization_Gating_Test extends TestCase {
 	/** @var string[] Handles registered elsewhere that the code under test only enqueues. */
 	private const BORROWED_SCRIPT_HANDLES = array( 'wp-jp-i18n-loader' );
 
+	/** @var array The `current_screen` callback `maybe_load_wp_build()` adds. */
+	private const SCREEN_ALIAS_CALLBACK = array( Jetpack_Backup::class, 'alias_screen_id_for_wp_build' );
+
+	/** @var array The `admin_print_scripts` callback `maybe_load_wp_build()` adds. */
+	private const INITIAL_STATE_CALLBACK = array( Jetpack_Backup::class, 'render_connection_initial_state' );
+
 	public function setUp(): void {
 		parent::setUp();
 
@@ -50,6 +59,7 @@ class Admin_Modernization_Gating_Test extends TestCase {
 		remove_all_filters( 'jetpack_offline_mode' );
 		remove_all_actions( 'load-jetpack_page_jetpack-backup' );
 		remove_all_actions( 'jetpack_use_iframe_authorization_flow' );
+		$this->leave_backup_admin_request();
 
 		$this->reset_scripts();
 		$this->set_admin_menu_items( array() );
@@ -139,11 +149,66 @@ class Admin_Modernization_Gating_Test extends TestCase {
 		$this->assertSame( 'VaultPress Backup', $items[0]['menu_title'] );
 	}
 
+	public function test_maybe_load_wp_build_does_nothing_when_not_modernized() {
+		$this->enter_backup_admin_request();
+
+		Jetpack_Backup::maybe_load_wp_build();
+
+		$this->assertFalse( has_action( 'current_screen', self::SCREEN_ALIAS_CALLBACK ) );
+		$this->assertFalse( has_action( 'admin_print_scripts', self::INITIAL_STATE_CALLBACK ) );
+	}
+
+	public function test_maybe_load_wp_build_does_nothing_off_the_backup_admin_page() {
+		add_filter( Jetpack_Backup::MODERNIZATION_FILTER, '__return_true' );
+		$this->enter_backup_admin_request();
+		$_GET['page'] = 'jetpack';
+
+		Jetpack_Backup::maybe_load_wp_build();
+
+		$this->assertFalse( has_action( 'current_screen', self::SCREEN_ALIAS_CALLBACK ) );
+		$this->assertFalse( has_action( 'admin_print_scripts', self::INITIAL_STATE_CALLBACK ) );
+	}
+
+	public function test_maybe_load_wp_build_hooks_the_page_when_modernized() {
+		add_filter( Jetpack_Backup::MODERNIZATION_FILTER, '__return_true' );
+		$this->enter_backup_admin_request();
+
+		Jetpack_Backup::maybe_load_wp_build();
+
+		$this->assertNotFalse( has_action( 'current_screen', self::SCREEN_ALIAS_CALLBACK ) );
+		$this->assertNotFalse( has_action( 'admin_print_scripts', self::INITIAL_STATE_CALLBACK ) );
+	}
+
+	public function test_render_connection_initial_state_emits_the_connection_global() {
+		ob_start();
+		Jetpack_Backup::render_connection_initial_state();
+		$output = ob_get_clean();
+
+		$this->assertStringContainsString( '<script id="jetpack-backup-connection-initial-state">', $output );
+		// render() declares a bare `var`; that is a window global only in a classic script tag.
+		$this->assertStringContainsString( 'JP_CONNECTION_INITIAL_STATE', $output );
+	}
+
 	/** `can_use_analytics()` needs offline mode off and the terms of service agreed. */
 	private function allow_analytics() {
 		add_filter( 'jetpack_offline_mode', '__return_false' );
 		Status_Cache::clear();
 		Jetpack_Options::update_option( 'tos_agreed', true );
+	}
+
+	/** `is_backup_admin_request()` reads `is_admin()` and `$_GET['page']`. */
+	private function enter_backup_admin_request() {
+		set_current_screen( 'jetpack_page_jetpack-backup' );
+		$_GET['page'] = Jetpack_Backup::JETPACK_BACKUP_SLUG;
+	}
+
+	/** Undo `enter_backup_admin_request()` and the hooks a successful load adds. */
+	private function leave_backup_admin_request() {
+		remove_action( 'current_screen', self::SCREEN_ALIAS_CALLBACK );
+		remove_action( 'admin_print_scripts', self::INITIAL_STATE_CALLBACK, 1 );
+
+		unset( $_GET['page'] );
+		set_current_screen( 'front' );
 	}
 
 	/** Undo every registration and enqueue these tests trigger. */

--- a/projects/packages/backup/tests/php/Admin_Modernization_Gating_Test.php
+++ b/projects/packages/backup/tests/php/Admin_Modernization_Gating_Test.php
@@ -86,6 +86,7 @@ class Admin_Modernization_Gating_Test extends TestCase {
 		Jetpack_Backup::enqueue_admin_scripts();
 
 		$this->assertFalse( wp_script_is( 'jetpack-backup', 'registered' ) );
+		$this->assertTrue( wp_script_is( 'wp-jp-i18n-loader', 'enqueued' ) );
 	}
 
 	public function test_enqueue_admin_scripts_enqueues_tracks_when_modernized() {

--- a/projects/packages/backup/tests/php/Admin_Modernization_Gating_Test.php
+++ b/projects/packages/backup/tests/php/Admin_Modernization_Gating_Test.php
@@ -1,0 +1,229 @@
+<?php
+/**
+ * Unit tests for the modernization gating around the admin UI.
+ *
+ * The companion of `Rest_Bridge_Gating_Test`, which covers the same guarantee
+ * for the REST routes: with the filter off, nothing about the admin page
+ * changes; with it on, the legacy assets stay off the page and the wp-build
+ * dashboard gets what it needs.
+ *
+ * @package automattic/jetpack-backup-plugin
+ */
+
+namespace Automattic\Jetpack\Backup\V0005;
+
+use Automattic\Jetpack\Admin_UI\Admin_Menu;
+use Automattic\Jetpack\Status\Cache as Status_Cache;
+use Jetpack_Options;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use WorDBless\Options as WorDBless_Options;
+use function add_filter;
+use function remove_all_actions;
+use function remove_all_filters;
+use function wp_deregister_script;
+use function wp_script_is;
+
+require_once __DIR__ . '/mock-wp-build-render-page.php';
+
+/**
+ * Tests the flag -> menu callback -> enqueue chain.
+ *
+ * @covers \Automattic\Jetpack\Backup\V0005\Jetpack_Backup
+ */
+#[CoversClass( Jetpack_Backup::class )]
+class Admin_Modernization_Gating_Test extends TestCase {
+
+	/**
+	 * Every script handle these tests may leave behind.
+	 *
+	 * @var string[]
+	 */
+	private const SCRIPT_HANDLES = array( 'jetpack-backup', 'jp-tracks', 'jp-tracks-functions' );
+
+	/**
+	 * Start from a clean script registry and an empty menu queue.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->reset_scripts();
+		$this->set_admin_menu_items( array() );
+	}
+
+	/**
+	 * Reset everything a test may have installed.
+	 */
+	public function tearDown(): void {
+		remove_all_filters( Jetpack_Backup::MODERNIZATION_FILTER );
+		remove_all_filters( 'jetpack_offline_mode' );
+		remove_all_actions( 'load-jetpack_page_jetpack-backup' );
+		remove_all_actions( 'jetpack_use_iframe_authorization_flow' );
+
+		$this->reset_scripts();
+		$this->set_admin_menu_items( array() );
+
+		Status_Cache::clear();
+		WorDBless_Options::init()->clear_options();
+
+		parent::tearDown();
+	}
+
+	/**
+	 * The flag is off unless a filter turns it on.
+	 */
+	public function test_is_modernized_defaults_to_false() {
+		$this->assertFalse( Jetpack_Backup::is_modernized() );
+	}
+
+	/**
+	 * The flag follows the filter in both directions.
+	 */
+	public function test_is_modernized_follows_the_filter() {
+		add_filter( Jetpack_Backup::MODERNIZATION_FILTER, '__return_true' );
+		$this->assertTrue( Jetpack_Backup::is_modernized() );
+
+		remove_all_filters( Jetpack_Backup::MODERNIZATION_FILTER );
+		add_filter( Jetpack_Backup::MODERNIZATION_FILTER, '__return_false' );
+		$this->assertFalse( Jetpack_Backup::is_modernized() );
+	}
+
+	/**
+	 * With the flag off, the legacy React bundle is still registered and
+	 * enqueued — the path the shipped plugin takes today.
+	 */
+	public function test_enqueue_admin_scripts_registers_legacy_script_when_not_modernized() {
+		Jetpack_Backup::enqueue_admin_scripts();
+
+		$this->assertTrue( wp_script_is( 'jetpack-backup', 'registered' ) );
+		$this->assertTrue( wp_script_is( 'jetpack-backup', 'enqueued' ) );
+	}
+
+	/**
+	 * With the flag on, the legacy bundle must not reach the page: wp-build
+	 * runs its own enqueue pipeline and would render a second dashboard.
+	 */
+	public function test_enqueue_admin_scripts_skips_legacy_script_when_modernized() {
+		add_filter( Jetpack_Backup::MODERNIZATION_FILTER, '__return_true' );
+
+		Jetpack_Backup::enqueue_admin_scripts();
+
+		$this->assertFalse( wp_script_is( 'jetpack-backup', 'registered' ) );
+	}
+
+	/**
+	 * The modernized page still gets the Tracks callables, so the dashboard
+	 * can record events. Without this the product's telemetry is blank.
+	 */
+	public function test_enqueue_admin_scripts_enqueues_tracks_when_modernized() {
+		$this->allow_analytics();
+		add_filter( Jetpack_Backup::MODERNIZATION_FILTER, '__return_true' );
+
+		$this->assertTrue( Jetpack_Backup::can_use_analytics() );
+
+		Jetpack_Backup::enqueue_admin_scripts();
+
+		$this->assertTrue( wp_script_is( 'jp-tracks', 'registered' ) );
+		$this->assertTrue( wp_script_is( 'jp-tracks-functions', 'enqueued' ) );
+	}
+
+	/**
+	 * The Tracks enqueue stays behind `can_use_analytics()`, same as the
+	 * legacy path: a site that has not agreed to the terms of service gets
+	 * no tracking script.
+	 */
+	public function test_enqueue_admin_scripts_skips_tracks_when_analytics_denied() {
+		add_filter( Jetpack_Backup::MODERNIZATION_FILTER, '__return_true' );
+
+		$this->assertFalse( Jetpack_Backup::can_use_analytics() );
+
+		Jetpack_Backup::enqueue_admin_scripts();
+
+		$this->assertFalse( wp_script_is( 'jp-tracks-functions', 'registered' ) );
+	}
+
+	/**
+	 * With the flag off, the menu points at the legacy settings page and
+	 * keeps the current labels.
+	 */
+	public function test_add_wp_admin_submenu_uses_legacy_callback_when_not_modernized() {
+		Jetpack_Backup::add_wp_admin_submenu();
+
+		$items = $this->get_admin_menu_items();
+		$this->assertCount( 1, $items );
+		$this->assertSame( Jetpack_Backup::JETPACK_BACKUP_SLUG, $items[0]['menu_slug'] );
+		$this->assertSame( array( Jetpack_Backup::class, 'plugin_settings_page' ), $items[0]['function'] );
+		$this->assertSame( 'Jetpack Backup', $items[0]['page_title'] );
+		$this->assertSame( 'Backup', $items[0]['menu_title'] );
+	}
+
+	/**
+	 * With the flag on, the same slug renders through the wp-build callback
+	 * and picks up the VaultPress Backup labels.
+	 */
+	public function test_add_wp_admin_submenu_uses_wp_build_callback_when_modernized() {
+		add_filter( Jetpack_Backup::MODERNIZATION_FILTER, '__return_true' );
+
+		Jetpack_Backup::add_wp_admin_submenu();
+
+		$items = $this->get_admin_menu_items();
+		$this->assertCount( 1, $items );
+		$this->assertSame( Jetpack_Backup::JETPACK_BACKUP_SLUG, $items[0]['menu_slug'] );
+		$this->assertSame( 'jetpack_backup_jetpack_backup_dashboard_wp_admin_render_page', $items[0]['function'] );
+		$this->assertSame( 'Jetpack VaultPress Backup', $items[0]['page_title'] );
+		$this->assertSame( 'VaultPress Backup', $items[0]['menu_title'] );
+	}
+
+	/**
+	 * Make `can_use_analytics()` return true: the site is not in offline
+	 * mode and has agreed to the terms of service.
+	 */
+	private function allow_analytics() {
+		add_filter( 'jetpack_offline_mode', '__return_false' );
+		Status_Cache::clear();
+		Jetpack_Options::update_option( 'tos_agreed', true );
+	}
+
+	/**
+	 * Deregister every script handle these tests touch.
+	 */
+	private function reset_scripts() {
+		foreach ( self::SCRIPT_HANDLES as $handle ) {
+			wp_deregister_script( $handle );
+		}
+	}
+
+	/**
+	 * Read the Admin_Menu package's queued menu items.
+	 *
+	 * @return array
+	 */
+	private function get_admin_menu_items() {
+		return $this->accessible_property( Admin_Menu::class, 'menu_items' )->getValue();
+	}
+
+	/**
+	 * Overwrite the Admin_Menu package's queued menu items.
+	 *
+	 * @param array $items The items to set.
+	 */
+	private function set_admin_menu_items( $items ) {
+		$this->accessible_property( Admin_Menu::class, 'menu_items' )->setValue( null, $items );
+	}
+
+	/**
+	 * Get an accessible reflection of a non-public property.
+	 *
+	 * @param string $class_name    The class owning the property.
+	 * @param string $property_name The property name.
+	 * @return \ReflectionProperty
+	 */
+	private function accessible_property( $class_name, $property_name ) {
+		$property = new \ReflectionProperty( $class_name, $property_name );
+		if ( \PHP_VERSION_ID < 80100 ) {
+			// Required to access non-public members before PHP 8.1; deprecated no-op since PHP 8.5.
+			$property->setAccessible( true );
+		}
+		return $property;
+	}
+}

--- a/projects/packages/backup/tests/php/mock-wp-build-render-page.php
+++ b/projects/packages/backup/tests/php/mock-wp-build-render-page.php
@@ -1,0 +1,20 @@
+<?php
+/**
+ * Stand-in for the render callback the wp-build dashboard generates.
+ *
+ * `Jetpack_Backup::add_wp_admin_submenu()` picks the wp-build callback only
+ * when `build/build.php` has been generated and loaded, which never happens
+ * in a unit test run. Defining the same function here lets the menu tests
+ * exercise the modernized branch of that choice.
+ *
+ * @package automattic/jetpack-backup-plugin
+ */
+
+if ( ! function_exists( 'jetpack_backup_jetpack_backup_dashboard_wp_admin_render_page' ) ) {
+	/**
+	 * Render the modernized Backup dashboard page. No-op in tests.
+	 *
+	 * @return void
+	 */
+	function jetpack_backup_jetpack_backup_dashboard_wp_admin_render_page() {}
+}

